services/thermal_mgr/thermal_mgr.c

```diff
@@ -41,19 +44,61 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 }
 
+
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL) return ERR_CODE_INVALID_ARG; 
+  if(thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_QUEUE_MSG;
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdPASS) return ERR_CODE_INVALID_QUEUE_MSG/*failed case*/;
```

Comment:
this function would only fail if the queue is full, not due to an invalid queue msg